plugins/inputs/procstat/connections_fallback.go

```diff
@@ -0,0 +1,46 @@
+//go:build !linux
+// +build !linux
+
+package procstat
+
+import (
+	"fmt"
+	"net"
+)
+
+type connInfo struct{}
+
+func (n *networkInfo) IsAListenPort(_ uint32) bool {
+	return false
+}
+
+func (n *networkInfo) Fetch() error {
+	return nil
+}
+
+func (n *networkInfo) GetConnectionsByPid(_ uint32) (conn []connInfo, err error) {
+	// Avoid "unused" errors
+	_ = dockerMACPrefix
+	_ = virtualBoxMACPrefix
+	_ = hardwareAddrLength
+	i := inodeInfo{}
+	_ = i.pid
+	_ = n.tcp
+	_ = n.listenPorts
+	_ = n.publicIPs
+	_ = n.privateIPs
+
+	return conn, fmt.Errorf("platform not supported")
+}
```

Comment:
Please use `errors.New` for errors without varadic arguments. Same for other locations in your code! Furthermore, I don't see why not just do
```suggestion
func (*networkInfo) GetConnectionsByPid(_ uint32) ([]connInfo, error) {
	return nil, errors.New("platform not supported")
}
```